pkg/api/store/cluster/cluster_store.go

```diff
@@ -633,6 +639,38 @@ func setBackupConfigSecretKeyIfNotExists(oldData, newData map[string]interface{}
 	}
 }
 
+func validateS3Credentials(data map[string]interface{}) error {
+	s3BackupConfig := values.GetValueN(data, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig", "s3BackupConfig")
+	if s3BackupConfig == nil {
+		return fmt.Errorf("Can't find S3 backup target configuration")
```

Comment:
Do we need to return an error here or returning nil is fine since enabling s3 backups isn't mandatory?